Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -623,8 +624,10 @@ lemma vanishingIdeal_irreducibleComponents :
     vanishingIdeal '' (irreducibleComponents <| PrimeSpectrum R) =
     minimalPrimes R := by
   rw [irreducibleComponents_eq_maximals_closed, minimalPrimes_eq_minimals,
-    ← minimals_swap, ← vanishingIdeal_isClosed_isIrreducible, image_minimals_of_rel_iff_rel]
-  exact fun s t hs _ ↦ vanishingIdeal_anti_mono_iff hs.1
+    image_antitone_setOf_maximal (fun s t hs _ ↦ (vanishingIdeal_anti_mono_iff hs.1).symm),
+    show Ideal.IsPrime = fun x ↦ x ∈ {x | Ideal.IsPrime x} from rfl,
```

Comment:
This `show` is not pretty. You could do: `← funext (@Set.mem_setOf_eq _ · Ideal.IsPrime)`, but I think it would be better to add the `rfl` lemma:
```lean
import Mathlib.Data.Set.Basic

example {α : Type u} {p : α → Prop} :
    (fun a ↦ a ∈ {x : α | p x}) = p :=
  rfl
```

Reply:
I opted for the former, because it looks to me like `mem_setOf_eq` should really be phrased as the lemma you suggest in the first place, and it wouldn't be the right place here to make that change.